File: tests/fixtures/closure_capture_js/handler.js
// Phase 8.1 regression: an arrow function captures a tainted variable
// from its enclosing scope and later sinks it via child_process.exec.
//
// The engine must follow the closure boundary — i.e. recognise that the
// inner arrow references `tainted` from `makeHandler` — and surface a
// taint-unsanitised-flow finding from env to exec.
function makeHandler() {
    const tainted = process.env.USER_INPUT;
    return (req) => {
        require('child_process').exec(tainted);
    };
}

const h = makeHandler();
h({});

module.exports = { makeHandler };

File: tests/fixtures/closure_capture_py/handler.py
# Regression fixture: a nested def captures a tainted variable from
# its enclosing function and sinks it via subprocess.run(..., shell=True).
#
# The engine must follow the closure boundary — i.e. recognise that
# `handler` references `tainted` from `make_handler` — and surface a
# taint-unsanitised-flow finding from env to subprocess.
import os
import subprocess


def make_handler():
    tainted = os.environ["USER_INPUT"]

    def handler(req):
        subprocess.run(tainted, shell=True)

    return handler


h = make_handler()
h({})
